seed/tests/test_analysis_pipelines.py

```diff
@@ -1001,32 +1001,115 @@ def test_invalid_meters(self):
         })
 
-    def test_valid_meters(self):
+    def test_valid_meters_use_most_recent_year(self):
         MeterReading.objects.filter(meter=self.meter).delete()
+        meter_readings = []
         for j in range(1, 13):
-            MeterReading.objects.create(
+            meter_reading = MeterReading.objects.create(
+                meter=self.meter,
+                start_time=make_aware(datetime(2020, j, 1, 0, 0, 0), timezone=self.timezone_object),
+                end_time=make_aware(datetime(2020, j, 28, 0, 0, 0), timezone=self.timezone_object),
+                reading=12345,
+                source_unit='kWh',
+                conversion_factor=1.00
+            )
+            meter_readings.append(SimpleMeterReading(meter_reading.start_time, meter_reading.end_time, meter_reading.reading))
+
+        # this meter should not be included in the results as its not in the year
+        MeterReading.objects.create(
+            meter=self.meter,
+            start_time=make_aware(datetime(2019, 1, 1, 0, 0, 0), timezone=self.timezone_object),
+            end_time=make_aware(datetime(2019, 1, 28, 0, 0, 0), timezone=self.timezone_object),
+            reading=12345,
+            source_unit='kWh',
+            conversion_factor=1.00
+        )
+
+        self.maxDiff = None
+        meter_readings_by_property_view, errors_by_property_view_id = _get_valid_meters([self.property_view.id], {"select_meters": "all"})
+        self.assertDictEqual(errors_by_property_view_id, {})
+        self.assertDictEqual(meter_readings_by_property_view, {self.property_view.id: meter_readings})
+
+    def test_valid_meters_use_custom_range(self):
+        MeterReading.objects.filter(meter=self.meter).delete()
+        meter_readings = []
+        for j in range(1, 12):
+            meter_reading = MeterReading.objects.create(
+                meter=self.meter,
+                start_time=make_aware(datetime(2020, j, 1, 0, 0, 0), timezone=self.timezone_object),
+                end_time=make_aware(datetime(2020, j, 28, 0, 0, 0), timezone=self.timezone_object),
+                reading=12345,
+                source_unit='kWh',
+                conversion_factor=1.00
+            )
+            meter_readings.append(SimpleMeterReading(meter_reading.start_time, meter_reading.end_time, meter_reading.reading))
+
+        # this meter should not be included in the results as its not in the range
+        MeterReading.objects.create(
+            meter=self.meter,
+            start_time=make_aware(datetime(2020, 12, 1, 0, 0, 0), timezone=self.timezone_object),
+            end_time=make_aware(datetime(2020, 12, 28, 0, 0, 0), timezone=self.timezone_object),
+            reading=12345,
+            source_unit='kWh',
+            conversion_factor=1.00
+        )
+
+        self.maxDiff = None
+        config = {
+            "select_meters": "date_range",
+            "meter": {
```

Comment:
When I select date range I only see "select_meters = date_range" but not the start and end dates. Is that intentional? It would be good to show the start and end dates. See image below.

![image](https://user-images.githubusercontent.com/6326294/217088959-458ea373-07e6-489a-aa15-4a5f4430855e.png)

Reply:
I did not alter how the analysis are displayed, let me see if I can add it.

Reply:
Sounds good. Thank you!

Reply:
This looks good now. However, can we have start_date first then end_date? I tried to figure out how to change this but couldn't figure it out quickly.